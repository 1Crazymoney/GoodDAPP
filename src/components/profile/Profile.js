import React from 'react'
import { StyleSheet, View } from 'react-native'
import Icon from 'react-native-elements/src/icons/Icon'
import { createStackNavigator } from '../appNavigation/stackNavigation'
import { Wrapper, Section, UserAvatar } from '../common'
import logger from '../../lib/logger/pino-logger'
import GDStore from '../../lib/undux/GDStore'
import EditProfile from './EditProfile'
import ProfileDataTable from './ProfileDataTable'

const log = logger.child({ from: 'Profile' })

const EditIcon = props => <IconButton {...props} name="edit" />
const PrivateIcon = props => <IconButton {...props} name="visibility" />

const IconButton = ({ onPress, disabled, ...iconProps }) => (
  <View style={styles.icon}>
    <Icon onPress={onPress} raised color="rgb(85, 85, 85)" {...iconProps} />
  </View>
)

const Profile = props => {
  const profile = GDStore.useStore().get('profile')
  const { screenProps } = props
  return (
    <Wrapper>
      <Section style={styles.section}>
        <Section.Row style={styles.centered}>
          <PrivateIcon onPress={() => log.debug('PrivateIcon')} />
          <UserAvatar profile={profile} />
          <EditIcon onPress={() => screenProps.push('EditProfile')} />
        </Section.Row>
        <ProfileDataTable profile={profile} />
      </Section>
    </Wrapper>
  )
}
const styles = StyleSheet.create({
  centered: {
    justifyContent: 'center',
    alignItems: 'flex-start'
  },
  section: {
    paddingLeft: '1em',
    paddingRight: '1em',
    marginBottom: 'auto',
    minHeight: '100%'
  },
  icon: {
    cursor: 'pointer'
  }
})

export default createStackNavigator({ Profile, EditProfile })
