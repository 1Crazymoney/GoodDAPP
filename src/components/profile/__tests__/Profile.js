import React from 'react'
import Profile from '../Profile'
import { createSwitchNavigator } from '@react-navigation/core'
import { createBrowserApp } from '@react-navigation/web'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('Profile', () => {
  it('renders without errors', () => {
    const WebRouter = createBrowserApp(createSwitchNavigator({ Profile }))

    const tree = renderer.create(<WebRouter />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const WebRouter = createBrowserApp(createSwitchNavigator({ Profile }))

    const component = renderer.create(<WebRouter />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
