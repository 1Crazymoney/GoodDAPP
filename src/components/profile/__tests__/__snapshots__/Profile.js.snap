// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile matches snapshot 1`] = `
<div
  style={
    Object {
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "horizontal",
      "WebkitBoxPack": "center",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "row",
      "WebkitFlexShrink": 0,
      "WebkitJustifyContent": "center",
      "alignItems": "center",
      "backgroundColor": "rgba(144,144,144,1.00)",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "display": "flex",
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexShrink": 0,
      "height": "60px",
      "justifyContent": "center",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "center",
      "msFlexDirection": "row",
      "msFlexNegative": 0,
      "msFlexPack": "center",
      "msFlexPreferredSize": "auto",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    style={
      Object {
        "WebkitAlignItems": "stretch",
        "WebkitBoxAlign": "stretch",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexBasis": "auto",
        "WebkitFlexDirection": "column",
        "WebkitFlexShrink": 0,
        "alignItems": "stretch",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0px",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0px",
        "borderRightStyle": "solid",
        "borderRightWidth": "0px",
        "borderTopStyle": "solid",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "display": "flex",
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexShrink": 0,
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "minHeight": "0px",
        "minWidth": "40px",
        "msFlexAlign": "stretch",
        "msFlexDirection": "column",
        "msFlexNegative": 0,
        "msFlexPreferredSize": "auto",
        "paddingBottom": "10px",
        "paddingLeft": "10px",
        "paddingRight": "10px",
        "paddingTop": "10px",
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "WebkitAlignItems": "stretch",
          "WebkitBoxAlign": "stretch",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "vertical",
          "WebkitFlexBasis": "auto",
          "WebkitFlexDirection": "column",
          "WebkitFlexShrink": 0,
          "alignItems": "stretch",
          "backgroundImage": "url(backButton.png)",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "auto",
          "flexDirection": "column",
          "flexShrink": 0,
          "height": "25px",
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "minHeight": "0px",
          "minWidth": "0px",
          "msFlexAlign": "stretch",
          "msFlexDirection": "column",
          "msFlexNegative": 0,
          "msFlexPreferredSize": "auto",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
          "position": "relative",
          "width": "25px",
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    dir="auto"
    style={
      Object {
        "WebkitBoxFlex": 2,
        "WebkitFlexGrow": 2,
        "borderBottomWidth": "0px",
        "borderLeftWidth": "0px",
        "borderRightWidth": "0px",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "color": "rgba(210,210,210,1.00)",
        "display": "inline",
        "flexGrow": 2,
        "fontFamily": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Ubuntu, \\"Helvetica Neue\\", sans-serif",
        "fontSize": "18px",
        "fontStyle": "inherit",
        "fontVariant": "inherit",
        "fontWeight": "inherit",
        "lineHeight": "inherit",
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "msFlexPositive": 2,
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "textAlign": "center",
        "textDecoration": "none",
        "textTransform": "uppercase",
        "whiteSpace": "pre-wrap",
        "wordWrap": "break-word",
      }
    }
  >
    Your Profile
  </div>
  <div
    style={
      Object {
        "WebkitAlignItems": "stretch",
        "WebkitBoxAlign": "stretch",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexBasis": "auto",
        "WebkitFlexDirection": "column",
        "WebkitFlexShrink": 0,
        "alignItems": "stretch",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0px",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0px",
        "borderRightStyle": "solid",
        "borderRightWidth": "0px",
        "borderTopStyle": "solid",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "display": "flex",
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexShrink": 0,
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "minHeight": "0px",
        "minWidth": "40px",
        "msFlexAlign": "stretch",
        "msFlexDirection": "column",
        "msFlexNegative": 0,
        "msFlexPreferredSize": "auto",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "position": "relative",
        "zIndex": 0,
      }
    }
  />
</div>
`;
